### PR TITLE
vscode-langservers-extracted: avoid vscodium runtime dependencies by …

### PR DESCRIPTION
**Pull request by SuperSandro2000**
…using its src

## Description of changes

```
 ➜ diffoscope result-*
--- result-after
+++ result-before
├── stat {}
│ @@ -1,7 +1,7 @@
│
│    Size: 79           Blocks: 8          IO Block: 4096   symbolic link
│  Device: 8,3  Access: (0777/lrwxrwxrwx)  Uid: ( 1000/  sandro)   Gid: (  100/   users)
│
│ -Modify: 2024-07-12 14:30:31.973660833 +0000
│ +Modify: 2024-07-12 14:30:53.818716131 +0000
│   --- result-after/bin
├── +++ result-before/bin
│ │   --- result-after/bin/vscode-css-language-server
│ ├── +++ result-before/bin/vscode-css-language-server
│ │ @@ -1,2 +1,2 @@
│ │  #! /nix/store/m101dg80ngyjdb02g6jwy80sr7kzj26g-bash-5.2p26/bin/bash -e
│ │ +exec "/nix/store/ilkfhnqz4xczrliqjva8770x2svbfznd-nodejs-20.14.0/bin/node"  /nix/store/jd0lrssiwfkgphbzi051php02lzwy72m-vscode-langservers-extracted-4.10.0/lib/node_modules/vscode-langservers-extracted/bin/vscode-css-language-server "$@"
│ │ -exec "/nix/store/ilkfhnqz4xczrliqjva8770x2svbfznd-nodejs-20.14.0/bin/node"  /nix/store/4m8sg5hjg7wlmflq8q6gj0i49qp8ladb-vscode-langservers-extracted-4.10.0/lib/node_modules/vscode-langservers-extracted/bin/vscode-css-language-server "$@"
│ │   --- result-after/bin/vscode-eslint-language-server
│ ├── +++ result-before/bin/vscode-eslint-language-server
│ │ @@ -1,2 +1,2 @@
│ │  #! /nix/store/m101dg80ngyjdb02g6jwy80sr7kzj26g-bash-5.2p26/bin/bash -e
│ │ +exec "/nix/store/ilkfhnqz4xczrliqjva8770x2svbfznd-nodejs-20.14.0/bin/node"  /nix/store/jd0lrssiwfkgphbzi051php02lzwy72m-vscode-langservers-extracted-4.10.0/lib/node_modules/vscode-langservers-extracted/bin/vscode-eslint-language-server "$@"
│ │ -exec "/nix/store/ilkfhnqz4xczrliqjva8770x2svbfznd-nodejs-20.14.0/bin/node"  /nix/store/4m8sg5hjg7wlmflq8q6gj0i49qp8ladb-vscode-langservers-extracted-4.10.0/lib/node_modules/vscode-langservers-extracted/bin/vscode-eslint-language-server "$@"
│ │   --- result-after/bin/vscode-html-language-server
│ ├── +++ result-before/bin/vscode-html-language-server
│ │ @@ -1,2 +1,2 @@
│ │  #! /nix/store/m101dg80ngyjdb02g6jwy80sr7kzj26g-bash-5.2p26/bin/bash -e
│ │ +exec "/nix/store/ilkfhnqz4xczrliqjva8770x2svbfznd-nodejs-20.14.0/bin/node"  /nix/store/jd0lrssiwfkgphbzi051php02lzwy72m-vscode-langservers-extracted-4.10.0/lib/node_modules/vscode-langservers-extracted/bin/vscode-html-language-server "$@"
│ │ -exec "/nix/store/ilkfhnqz4xczrliqjva8770x2svbfznd-nodejs-20.14.0/bin/node"  /nix/store/4m8sg5hjg7wlmflq8q6gj0i49qp8ladb-vscode-langservers-extracted-4.10.0/lib/node_modules/vscode-langservers-extracted/bin/vscode-html-language-server "$@"
│ │   --- result-after/bin/vscode-json-language-server
│ ├── +++ result-before/bin/vscode-json-language-server
│ │ @@ -1,2 +1,2 @@
│ │  #! /nix/store/m101dg80ngyjdb02g6jwy80sr7kzj26g-bash-5.2p26/bin/bash -e
│ │ +exec "/nix/store/ilkfhnqz4xczrliqjva8770x2svbfznd-nodejs-20.14.0/bin/node"  /nix/store/jd0lrssiwfkgphbzi051php02lzwy72m-vscode-langservers-extracted-4.10.0/lib/node_modules/vscode-langservers-extracted/bin/vscode-json-language-server "$@"
│ │ -exec "/nix/store/ilkfhnqz4xczrliqjva8770x2svbfznd-nodejs-20.14.0/bin/node"  /nix/store/4m8sg5hjg7wlmflq8q6gj0i49qp8ladb-vscode-langservers-extracted-4.10.0/lib/node_modules/vscode-langservers-extracted/bin/vscode-json-language-server "$@"
│ │   --- result-after/bin/vscode-markdown-language-server
│ ├── +++ result-before/bin/vscode-markdown-language-server
│ │ @@ -1,2 +1,2 @@
│ │  #! /nix/store/m101dg80ngyjdb02g6jwy80sr7kzj26g-bash-5.2p26/bin/bash -e
│ │ +exec "/nix/store/ilkfhnqz4xczrliqjva8770x2svbfznd-nodejs-20.14.0/bin/node"  /nix/store/jd0lrssiwfkgphbzi051php02lzwy72m-vscode-langservers-extracted-4.10.0/lib/node_modules/vscode-langservers-extracted/bin/vscode-markdown-language-server "$@"
│ │ -exec "/nix/store/ilkfhnqz4xczrliqjva8770x2svbfznd-nodejs-20.14.0/bin/node"  /nix/store/4m8sg5hjg7wlmflq8q6gj0i49qp8ladb-vscode-langservers-extracted-4.10.0/lib/node_modules/vscode-langservers-extracted/bin/vscode-markdown-language-server "$@"
```

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Lord-Valen** [comment]
This fails on x86 darwin
```
> Running phase: unpackPhase

> unpacking source archive /nix/store/036vik9b785i8i45h9yrnljzarzwbds9-source

> calling 'unpackCmd' function hook '_defaultUnpack' /nix/store/036vik9b785i8i45h9yrnljzarzwbds9-source

> unpacking source archive /nix/store/kzbrwnbmbahiyj11ik21hqmbkan3k71b-VSCodium-darwin-x64-1.90.2.24171.zip

> calling 'unpackCmd' function hook '_defaultUnpack' /nix/store/kzbrwnbmbahiyj11ik21hqmbkan3k71b-VSCodium-darwin-x64-1.90.2.24171.zip

> do not know how to unpack source archive /nix/store/kzbrwnbmbahiyj11ik21hqmbkan3k71b-VSCodium-darwin-x64-1.90.2.24171.zip
```

**Comment by Lord-Valen** [comment]
Adding `unzip` to `nativeBuildInputs` when building on Darwin should be sufficient.

**Comment by SuperSandro2000** [comment]
@ofborg build vscode-langservers-extracted

**Comment by SuperSandro2000** [comment]
@Lord-Valen can you test darwin again? ofborg is taking its good time

**Comment by Lord-Valen** [comment]
I don't have a darwin machine. I was waiting on ofborg myself. Maybe @malob can help?

**Comment by malob** [comment]
The path you're looking for is:

```
../VSCodium.app/Contents/Resources/app/extensions
```

**Comment by SuperSandro2000** [comment]
@NixOS/darwin-maintainers could you quickly test this?

**Comment by Mastermindaxe** [comment]
This fails on `aarch64-darwin`:
```
last 10 log lines:
       > rebuilt dependencies successfully
       > patching script interpreter paths in node_modules
       > Finished npmConfigHook
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > calling 'preConfigure' function hook '_multioutConfig'
       > no configure script, doing nothing
       > Running phase: buildPhase
       > babel:
       >   ../Contents/Resources/app/extensions/css-language-features/server/dist/node does not exist
```

File: pkgs/development/tools/language-servers/vscode-langservers-extracted/default.nix [hunk 1]
```diff
@@ -1,24 +1,31 @@
...
 
+  nativeBuildInputs = [ unzip ];
+
   buildPhase =
     let
       extensions =
         if stdenv.isDarwin
-        then "${vscodium}/Applications/VSCodium.app/Contents/Resources/app/extensions"
-        else "${vscodium}/lib/vscode/resources/app/extensions";
+        then "../Contents/Resources/app/extensions"
```
Comment (SuperSandro2000) [inline pkgs/development/tools/language-servers/vscode-langservers-extracted/default.nix:27]: @Mastermindaxe can you check if the bellow suggestion fixes it?

```suggestion
        then "../VSCodium.app/Contents/Resources/app/extensions"
```
or try to find out by using -K/--keep-failed what the correct paths are?

Comment (cideM) [reply pkgs/development/tools/language-servers/vscode-langservers-extracted/default.nix:27]: With this change it works on my `aarch64-darwin` machine

Comment (Mastermindaxe) [reply pkgs/development/tools/language-servers/vscode-langservers-extracted/default.nix:27]: Same 👍

Comment (SuperSandro2000) [reply pkgs/development/tools/language-servers/vscode-langservers-extracted/default.nix:27]: Nice! Thanks!

**Review by Lord-Valen — APPROVED** [review]
OfBorg just got broken but thanks to user testing on Linux and Darwin I think we can say this is good to go.

**Comment by jian-lin** [comment]
@ofborg eval

to trigger a new eval

**Pull request merged**